lib/tool-bar-view.coffee

```diff
@@ -61,6 +61,9 @@ module.exports = class ToolBarView extends View
     @.element.addEventListener 'scroll', @drawGutter
     window.addEventListener 'resize', @drawGutter
 
+    if atom.config.get 'tool-bar.automaticallyResizeIcons'
```

Comment:
Interesting issues here.  In the code below, the `atom.config.observe` part works just fine (log messages print appropriately), but adding and removing event listeners doesn't seem to "stick" and whether the event is listened to or not doesn't seem to change at all.

``` coffee
    atom.config.observe 'tool-bar.automaticallyResizeIcons', ->
      if atom.config.get 'tool-bar.automaticallyResizeIcons'
        console.log "Now it's turned on."
        window.addEventListener 'resize', @updateAutoSize
        @updateAutoSize
      else
        console.log "Now it's turned off."
        window.removeEventListener 'resize', @updateAutoSize
```

It would be easy enough to keep the event listener on all the time, and instead just silence the actual updating, but definitely not quite as nice. (And is there much overhead associated with having the event listener on?)

``` coffee
  updateAutoSize: =>
    if !atom.config.get 'tool-bar.automaticallyResizeIcons'
        return
    # else continue with resizing
```

I'm going to keep playing with it and try to find some similar examples, but I haven't done much coffee/javascript, so let me know if looks very wrong here.